crates/noirc_errors/src/position.rs

```diff
@@ -58,11 +58,15 @@ impl<T> std::borrow::Borrow<T> for Spanned<T> {
 pub struct Span(ByteSpan);
 
 impl Span {
-    pub fn new(range: Range<u32>) -> Span {
+    pub fn new(mut range: Range<u32>) -> Span {
+        if range.start > range.end {
```

Comment:
when does this happen?

Reply:
I think this is a bug in the new version of chumsky, it happens when combining ranges during parsing within some combinators now unfortunately.

Reply:
I've opened an issue for this here: https://github.com/noir-lang/noir/issues/156 so as not to block this PR